sherpa/optmethods/optscipy.py

```diff
@@ -0,0 +1,502 @@
+        cannot be set by the user. Below is a list of attributes most likely to be
+        be of interest for Sherpa users, see `scipy.optimize.shgo`
+        documentation for details of all parameters.
+
+        Attributes
+        ----------
+        n : int
+            Number of sampling points used in the construction of the simplicial complex.
+        iters : int
+            Number of iterations used in the construction of the simplicial complex.
+            Default is 1.
+        callback : callable
+            A callable called after each iteration.
+        minimizer_kwargs : dict
+            A dictionary of options to pass to the local minimizer.
+        options : dict
+            A dictionary of solver options. See the `scipy.optimize.shgo`
+            documentation for a full list.
+        sampling_method : str or function
+            Current built in sampling method options are `halton`, `sobol`
+            and `simplicial`.
+            See the `scipy.optimize.shgo` documentation for details.
+        workers : int or map-like callable
+            The number of workers to use for parallelization.
+        """
+        _scipy_func = staticmethod(optimize.shgo)
+        _requires_finite_bounds = True
+
+
+    class Scipy_Direct(ScipyBase):
+        """Optimizer using `scipy.optimize.direct`.
+
+        The `scipy.optimize.direct` function implements the DIRECT algorithm
+        for global optimization.
+
+        Sherpa will automatically convert statistics functions, input values,
+        parameter limits etc. to the format required by the scipy function and those
+        cannot be set by the user. Below is a list of attributes most likely to be
+        be of interest for Sherpa users, see `scipy.optimize.direct`
+        documentation for details of all parameters.
+
+        Attributes
+        ----------
+        eps : float
+            Minimal required difference between steps. Default is 1e-4.
+        maxfun : int or None
+            Approximate upper bound on objective function evaluations.
+        maxiter : int
+            Maximum number of iterations.
+        locally_biased : bool
+            If True (default), use the locally biased variant of the algorithm
+            known as DIRECT_L.
+        vol_tol : float
+            Terminate the optimization once the volume of the hyperrectangle
+            containing the lowest function value is smaller than vol_tol of the
+            complete search space. Must lie between 0 and 1. Default is 1e-16.
+        len_tol : float
+            Allowed range is 0 to 1. Default is 1e-6.
+        """
+        _scipy_func = staticmethod(optimize.direct)
+        _requires_finite_bounds = True
+
+
+    __all__.append('Scipy_Minimize')
+    __all__.append('Scipy_Basinhopping')
+    __all__.append('Scipy_DifferentialEvolution')
+    __all__.append('Scipy_DualAnnealing')
+    __all__.append('Scipy_Shgo')
+    __all__.append('Scipy_Direct')
```

Comment:
Would it not be better to do something like

```python
__all__.extend(['Scipy_Minimize', 'Scipy_Basinhoppnig', ...])
```

I'm not sure which approach I prefer for readability (I am not concerned about efficiency here).

Reply:
Either works, I don't care strongly either way.